src/gsy_e/models/strategy/state/heatpump_tank_states/all_tanks_state.py

```diff
@@ -31,19 +34,36 @@ class AllTanksState:
     def __init__(self, tank_parameters: List[TankParameters]):
         self.tanks_states = [heatpump_state_factory(tank) for tank in tank_parameters]
 
+    def _get_scaling_factors_for_charging_energy(self, time_slot):
+        _current_energy_demands_tanks = [
+            tank.get_heat_consumption_kJ(time_slot) for tank in self.tanks_states
```

Comment:
I think that this formula will lead to unbalanced tank temperatures / tank SOCs. I do not recall our discussion anymore, however did we also evaluate using the available heat capacity of the water tank - in other words, the max heat energy that can be stored in the water tank? That way, charging would be distributed proportionally to all water tanks. 

I vaguely recall that we discussed about this though, and we though that the more the consumption of a tank is, the more it should be charged, however I am not sure about that. Feel free to ignore the comment if we came to such a conclusion